tests/ClusterHealthTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+use Elasticsearch\Client;
+use Elasticsearch\Namespaces\ClusterNamespace;
+
+class ClusterHealthTest extends ElasticSearcherTestCase
+{
+	/**
+	 * @var PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $clientMock;
+
+	/**
+	 * @var PHPUnit_Framework_MockObject_MockObject|ClusterNamespace
+	 */
+	private $clusterMock;
+
+	public function setUp()
+	{
+		parent::setUp();
+
+		$this->clusterMock = $this->getMockBuilder(ClusterNamespace::class)
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->clientMock = $this->getMockBuilder(Client::class)
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->getElasticSearcher()->setClient($this->clientMock);
+	}
+
+	public function testCanCheckIfSingleNodeClusterIsNoHealthy()
+	{
+		$this->clusterMock->expects($this->once())
+			->method('health')
+			->willReturn([
+				'status' => 'red',
+				'number_of_nodes' => 1,
+			]);
+
+		$this->clientMock->expects($this->once())
+			->method('cluster')
+			->willReturn($this->clusterMock);
+
+		$this->assertFalse($this->getElasticSearcher()->isHealthy(), 'Expected cluster not to be healthy, but it is');
+	}
+
+	public function testCanCheckIfMultiNodeClusterIsHealthy()
+	{
+		$this->clusterMock->expects($this->once())
+			->method('health')
+			->willReturn([
+				'status' => 'green',
+				'number_of_nodes' => 2,
+			]);
+
+		$this->clientMock->expects($this->once())
+			->method('cluster')
+			->willReturn($this->clusterMock);
+
+		$this->assertTrue($this->getElasticSearcher()->isHealthy(), 'Expected cluster to be healthy, but it is not');
+	}
+
+	public function testCanCheckIfMultiNodeClusterIsNotHealthy()
```

Comment:
Can't all the tests be summarised to one test? Only 3 paths exists, the status is green, yellow or red.

The tests names don't make much sense anymore since the health check does not depend on the number of nodes.